src/python.rs

```diff
@@ -481,7 +481,24 @@ impl Taxonomy {
                     "Node can not be moved to its child",
                 ));
             }
+
+            let old_parent_idx = self.tax.parent_ids[idx];
+            let new_parent_idx = py_try!(self.tax.to_internal_index(p));
+
+            // update id to parent_id
             self.tax.parent_ids[idx] = py_try!(self.tax.to_internal_index(p));
+
+            // remove current node from children lookup for old parent
+            let removal_index = self.tax.children_lookup[old_parent_idx]
+                .binary_search(&idx)
+                .unwrap();
+            self.tax.children_lookup[old_parent_idx].remove(removal_index);
+
+            self.tax.children_lookup[old_parent_idx].sort_unstable();
```

Comment:
if `self.tax.children_lookup[old_parent_idx]` is a sorted vector (array) then removing an element will not make it unsorted.